Rectangle/WindowManager.swift

```diff
@@ -98,13 +98,19 @@ class WindowManager {
         }
 
         let resultingRect = frontmostWindowElement.rectOfElement()
-        windowHistory.lastRectangleActions[windowId] = RectangleAction(action: calcResult.resultingAction, rect: resultingRect)
+        let count = (lastRectangleAction?.action == calcResult.resultingAction) ? lastRectangleAction!.count + 1 : 1
```

Comment:
Again, I would prefer not to force unwrap, even though this is another scenario where lastRectangleAction would not be nil. 
Maybe something like this 

```
var count = 1
if let lastRectangleAction = lastRectangleAction {
    if lastRectangleAction.action = calcResult.resultingAction {
        count = lastRectangleAction.count + 1
    }
}
```

It's kind of annoying to have the `if let` assignment to sift through a potential nil, but it's not hard to get used to seeing it, and in my opinion it just beats force unwrapping. The reason why I'm so cautious about force unwrapping is that it's bit me several times before. When you're dealing with certain APIs, like a lot of C APIs (which we're not here), sometimes values can become nil without you knowing it. I had a lot of weird crashes in Multitouch and all of it came down to force unwrapping. Basically, my thought is just err on the side of no crashes.

Reply:
I look at your sample code and I just think it's overly verbose for what you're trying to do, 5 lines to safely unwrap a `Optional<>`, I really wish the language had the `.orElse()` Java Optional method (https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElse-T-)

Again what about:

`let count = (lastRectangleAction?.action == calcResult.resultingAction) ? (lastRectangleAction?.count ?? 0) + 1 : 1`

Reply:
I've realised you can extend the languages Optional, meaning a `getOrElse()` method (similar to Java .orElse() method) is possible! 

Might reduce the boiler-plate Swift Optional unwrapping

```
extension Optional  {
    func getOrElse<T>(_ value: T) -> T {
        if self != nil {
            return self as! T
        }
        return value
    }
}
```

Reply:
I'm good with your suggestion on providing default values with `??`.  

Although it's nice to be able to extend language functionality, I typically avoid it because it's one of those things that can get out of hand if overused. Switching to different projects and lugging around your suitcase of language extensions isn't ideal.